pkg/experimental/genrpm.bzl

```diff
@@ -0,0 +1,611 @@
+                        ),
+                        "data",
+                    )
+                else:
+                    dest_check_map[dest] = d.label
+
+    #### Procedurally-generated scripts/lists (%install, %files)
+
+    # Build up the install script
+    install_script_pieces = []
+    if ctx.attr.debug:
+        install_script_pieces.append("set -x")
+
+    # {0} is the source, {1} is the dest
+    install_file_stanza_fmt = """
+install -d %{{buildroot}}/$(dirname {1})
+cp -r {0} %{{buildroot}}/{1}
+    """
+
+    # {0} is the directory name
+    #
+    # This may not be strictly necessary, given that they'll be created in the
+    # CPIO when rpmbuild processes the `%files` list.
+    install_dir_stanza_fmt = """
+install -d %{{buildroot}}/{0}
+    """
+
+    # Build up the RPM files list (%files -f)
+    rpm_files_list = []
+
+    # Iterate over all incoming data, creating datasets as we go from the
+    # actual contents of the RPM.
+    #
+    # This is a naive approach to script creation is almost guaranteed to
+    # produce an installation script that is longer than necessary.  A better
+    # implementation would track directories that are created and ensure that
+    # they aren't unnecessarily recreated.
+    for elem in ctx.attr.data:
+        if PackageFileInfo in elem:
+            pfi = elem[PackageFileInfo]
+            file_base = "%attr({}) {}".format(
+                ", ".join(pfi.attrs["unix"]),
+                "%" + pfi.section if pfi.section else "",
+            )
+            for (source, dest) in zip(pfi.srcs, pfi.dests):
+                rpm_files_list.append(file_base + " /" + dest)
+
+                install_script_pieces.append(install_file_stanza_fmt.format(
+                    source.path,
+                    dest,
+                ))
+        if PackageDirInfo in elem:
+            pdi = elem[PackageDirInfo]
+            file_base = "%attr({}) {}".format(
+                ", ".join(pdi.attrs["unix"]),
+                "%" + pdi.section if pdi.section else "",
+            )
+            for d in pdi.dirs:
+                rpm_files_list.append(file_base + " /" + d)
+
+                install_script_pieces.append(install_dir_stanza_fmt.format(
+                    d,
+                ))
+
+    install_script = ctx.actions.declare_file("{}.spec.install".format(rpm_name))
+    ctx.actions.write(
+        install_script,
+        "\n".join(install_script_pieces),
+    )
+    files.append(install_script)
+    args.append("--install_script=" + install_script.path)
+
+    rpm_files_file = ctx.actions.declare_file(
+        "{}.spec.files".format(rpm_name),
+    )
+    ctx.actions.write(
+        rpm_files_file,
+        "\n".join(rpm_files_list),
+    )
+    files.append(rpm_files_file)
+    args.append("--file_list=" + rpm_files_file.path)
+
+    for f in ctx.files.data:
+        args.append(f.path)
+
+    #### Call the generator script.
+
+    # TODO(katre): Generate a source RPM.
```

Comment:
I don't think katre is signing up for this. And getting source RPMs is an entirely aspirational goal at this time.

Reply:
Sure thing; I'll remove this.  This is actually also in pkg/rpm.bzl too: https://github.com/bazelbuild/rules_pkg/blob/master/pkg/rpm.bzl#L92

Reply:
Removed in 229a244.